duckduckgo/iOS
# Sync Setup UI

Task/Issue URL: https://app.asana.com/0/72649045549333/1203756800930488/f
Tech Design URL:
CC:

**Description**:

Adds the setup UI for sync behind an internal user flag.

***Notes***:

* The "camera" screens are fixed in dark mode and on iPhone also fixed in portrait mode by design.
* Some of the code is hacky/place holder for future work

***Principles***:

I'm not precious about any of this, but it helped me to maintain some structure.

* Most of the code is in a local Swift Package which is accessed/bootstrapped by the view controller which lives in the app with the required assets. 
* The Swift Package has been carved up into folders by type (Views, ViewModels) with public stuff as near to the top level as possible.  
* The original intention was that the ViewModels be re-usable across platforms, but that will come later.  If it does the ViewModels will move into BSK, then we'll likely drop the "View" folder and move the structure up a level. 
* Each View swift files contains the `public` view plus any required helpers which are marked as private.   
* Any views that are re-usable and not intended to be public have been extracted into their own files and placed in the `Views\internal` folder with default access.
* I've tried to minimise the UI code in the body var and extract things into view build functions where possible/logical.  Please call out anything that looks like it could be clearer.




**Steps to test this PR**:
1. Run as debug or login to the DDG endpoint
2. Ensure Sync item appears in Settings
3. Smoke test the UI
4. If some code looks hacky and it's not obvious it will be fixed later, please call it out (might just need comment to that effect)



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
